### PR TITLE
add some and many for Lazy List

### LINKED ISSUE #112: add some many to Lazy List
There is no definition of some and many for Lazy List.
Is it intentional? (why)
If it's not i could open PR

### PR DESCRIPTION
**Pull request by safareli**
fix #112

### CODE REVIEW

**Comment by garyb** [comment]
Looks good to me :+1:

File: src/Data/List/Lazy.purs [hunk 1]
```diff
@@ -160,6 +162,21 @@ replicateM n m
...
+-- |
+-- | The `Lazy` constraint is used to generate the result lazily, to ensure
+-- | termination.
+some :: forall f a. Alternative f => Z.Lazy (f (List  a)) => f a -> f (List  a)
+some v = cons <$> v <*> Z.defer (\_ -> many v)
+
+-- | Attempt a computation multiple times, returning as many successful results
+-- | as possible (possibly zero).
+-- |
+-- | The `Lazy` constraint is used to generate the result lazily, to ensure
```
Comment (paf31) [inline src/Data/List/Lazy.purs]: There is no `Lazy` constraint here.

**Comment by paf31** [comment]
Thanks!

File: src/Data/List/Lazy.purs (hunk 1 above)
Comment (safareli) [inline src/Data/List/Lazy.purs]: Copied it from [List.purs](https://github.com/purescript/purescript-lists/blob/master/src/Data/List.purs#L158).  But even tho `Lazy` constraint isn't used in `many`, it's used in `some`, which is used by `many`, so it's technically true.  If want better wording here please let me know and i'll update it in both places

Comment (paf31) [inline src/Data/List/Lazy.purs]: Sorry, I misread it, please ignore my comment.

File: src/Data/List/Lazy.purs [hunk 2]
```diff
@@ -160,6 +162,21 @@ replicateM n m
       as <- replicateM (n - one) m
       pure (cons a as)
 
+-- | Attempt a computation multiple times, requiring at least one success.
+-- |
+-- | The `Lazy` constraint is used to generate the result lazily, to ensure
+-- | termination.
+some :: forall f a. Alternative f => Z.Lazy (f (List  a)) => f a -> f (List  a)
+some v = cons <$> v <*> Z.defer (\_ -> many v)
```
Comment (paf31) [inline src/Data/List/Lazy.purs]: Is it possible to generate the tail lazily here?

Comment (safareli) [inline src/Data/List/Lazy.purs]: That's what `Z.defer` is doing here.

Comment (paf31) [inline src/Data/List/Lazy.purs]: Yes, I think this is fine. Sorry again :)

Comment (safareli) [inline src/Data/List/Lazy.purs]: It's fine :p

**Pull request merged**